platform/linux-generic/odp_timer.c

```diff
@@ -581,14 +581,20 @@ static odp_event_t timer_set_unused(timer_pool_t *tp, uint32_t idx)
 	return old_event;
 }
 
-static inline odp_event_t timer_free(timer_pool_t *tp, uint32_t idx)
+int odp_timer_free(odp_timer_t hdl)
 {
+	timer_pool_t *tp = handle_to_tp(hdl);
+	uint32_t idx = handle_to_idx(hdl, tp);
 	_odp_timer_t *tim = &tp->timers[idx];
 	tick_buf_t *tb = &tp->tick_buf[idx];
 
 	/* Free the timer by setting timer state to unused and
 	 * grab any timeout event */
 	odp_event_t old_event = timer_set_unused(tp, idx);
+	if (old_event != ODP_EVENT_INVALID) {
+		_ODP_ERR("Timer is active\n");
+		return -1;
+	}
```

Comment:
Aren't we leaking an event here? Would it be better to not make any changes if the timer is active?

Reply:
Yes, but that should not never happen after this API change. We can simplify / rewrite (entire timer) implementation after this API change and removal of deprecated timer set functions are merged into repo.